File: app/src/main/java/com/recursia/popularmovies/presentation/views/IntroScreenContract.java
package com.recursia.popularmovies.presentation.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.recursia.popularmovies.presentation.models.SectionItem;

import java.util.List;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface IntroScreenContract extends MvpView {

    void openMainScreen();

    void setNextSection();

    void setPreviousSection();

    //Next button
    void showNextButton();

    void hideNextButton();

    //Prev button
    void showPreviousButton();

    void hidePreviousButton();

    //Finish button
    void showFinishButton();

    void hideFinishButton();

    //View pager
    void setViewPagerData(List<SectionItem> sectionItems);

}

File: app/src/main/java/com/recursia/popularmovies/presentation/views/contracts/IntroScreenContract.kt
package com.recursia.popularmovies.presentation.views.contracts

import com.recursia.popularmovies.presentation.models.SectionItem
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface IntroScreenContract : MvpView {

    fun setNextSection()

    fun setPreviousSection()

    // Next button
    fun showNextButton()

    fun hideNextButton()

    // Prev button
    fun showPreviousButton()

    fun hidePreviousButton()

    // Finish button
    fun showFinishButton()

    fun hideFinishButton()

    // View pager
    fun setViewPagerData(sectionItems: List<SectionItem>)
}
